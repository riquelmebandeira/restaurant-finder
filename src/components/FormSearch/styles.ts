import styled, { css } from 'styled-components'

export const Form = styled.form`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    max-width: 54rem;
    height: 5rem;
    padding: 1.3rem 2rem;
    border: 1px solid ${theme.colors.lightMedium};
    border-radius: 0.8rem;
    display: flex;
    align-items: center;

    input {
      border: none;
      margin-left: 1rem;
      outline: none;
      width: 100%;

      &::placeholder {
        font-size: ${theme.font.sizes.xsmall};
        font-family: ${theme.font.family.default};
        color: ${theme.colors.dark};
      }
    }
  `}
`
