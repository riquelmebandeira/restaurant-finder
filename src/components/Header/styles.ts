import styled from 'styled-components'
import img from '../../assets/header-background.png'

export const Header = styled.header`
  background-image: url(${img});
  background-size: cover;
  background-position: center;
  height: 37.5rem;
  position: relative;
  padding: 0 1rem;
  width: 100%;

  .content {
    max-width: 110rem;
    display: flex;
    justify-content: space-between;
    margin: auto;
    padding: 3rem 1rem;
  }

  .text-box {
    margin-top: -0.9rem;
    text-align: right;
  }

  form {
    bottom: 4rem;
    left: 50%;
    position: absolute;
    transform: translateX(-50%);
    width: 95%;
  }
`
