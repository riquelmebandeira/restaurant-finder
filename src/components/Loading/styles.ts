import styled, { keyframes } from 'styled-components'
import loadingIcon from '../../assets/icons/loading.svg'

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg)
  }
`

export const Loading = styled.img.attrs({
  alt: 'Ã­cone de carregamento',
  src: loadingIcon
})`
  animation: ${rotate} 2s linear infinite;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
