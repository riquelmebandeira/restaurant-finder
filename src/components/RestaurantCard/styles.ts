import styled from 'styled-components'

export const Card = styled.div`
  width: 35rem;
  height: 17rem;
  position: relative;
  border-radius: 0.8rem;

  *:not(h6) {
    width: 100%;
    height: 100%;
  }

  img {
    object-fit: cover;
    border-radius: 0.8rem;
    filter: brightness(60%);
  }

  h6 {
    position: absolute;
    bottom: 1rem;
    left: 1rem;
    color: ${({ theme }) => theme.colors.white};
  }
`
