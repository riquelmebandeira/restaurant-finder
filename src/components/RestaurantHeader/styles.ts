import styled from 'styled-components'

type HeaderProps = {
  backgroundImg: string
}

export const Header = styled.header<HeaderProps>`
  position: relative;
  height: 19rem;
  width: 100%;

  // BACKGROUND IMAGE
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url(${(props) => props.backgroundImg});
    background-size: cover;
    background-position: center;
    filter: brightness(50%);
  }
`

export const BackButton = styled.div`
  display: flex;
  cursor: pointer;
  position: absolute;
  top: 5.2rem;
  left: 17.4rem;
  z-index: 100;

  img {
    margin-right: 1.4rem;
  }
`

export const Container = styled.div`
  align-items: center;
  display: flex;
  margin: auto;
  max-width: 55rem;
  padding: 0 1rem;
  position: relative;
  top: 4rem;
  z-index: 100;

  img {
    border-radius: 50%;
    height: 12rem;
    margin-right: 2.5rem;
    width: 12rem;
  }
`
