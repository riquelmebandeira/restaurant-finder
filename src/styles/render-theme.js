import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from './theme'

export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export const renderWithRouterAndTheme = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </ThemeProvider>
  )
}
