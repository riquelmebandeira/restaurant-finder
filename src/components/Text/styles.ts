import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'

type TextProps = {
  color?: keyof typeof theme.colors
  fontWeight?: 100 | 400 | 800
  fontSize?: keyof typeof theme.font.sizes
}

export const Text = styled.p<TextProps>`
  ${({ color = 'white', fontWeight = 400, fontSize = 'xsmall' }) => css`
    font-weight: ${fontWeight};
    color: ${theme.colors[color]};
    font-size: ${theme.font.sizes[fontSize]};
  `}
`
