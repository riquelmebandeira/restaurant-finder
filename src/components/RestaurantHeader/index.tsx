import * as S from './styles'
import leftArrow from '../../assets/icons/left-arrow.svg'
import { Heading } from '../Heading/styles'
import { Text } from '../Text/styles'
import { useNavigate } from 'react-router-dom'

type RestaurantHeaderProps = {
  name: string
  image: string
  logo: string
  telephone: string
  website: string
}

const RestaurantHeader = ({
  name,
  image,
  logo,
  telephone,
  website
}: RestaurantHeaderProps) => {
  const navigate = useNavigate()

  return (
    <S.Header backgroundImg={image}>
      <S.BackButton onClick={() => navigate(-1)}>
        <img
          src={leftArrow}
          alt="botÃ£o de voltar"
        />
        <Text
          as="span"
          color="white"
          fontSize="xxsmall"
          fontWeight={800}
        >
          Voltar
        </Text>
      </S.BackButton>

      <S.Container>
        <img
          src={logo}
          alt={`logomarca do restaurante ${name}`}
        />
        <div>
          <Heading>{name}</Heading>
          <Text>
            {telephone}
            <br />
            {website}
          </Text>
        </div>
      </S.Container>
    </S.Header>
  )
}

export default RestaurantHeader
