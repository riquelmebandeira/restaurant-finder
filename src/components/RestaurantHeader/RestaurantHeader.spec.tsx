import { renderWithRouterAndTheme } from '../../styles/render-theme'
import RestaurantHeader from '.'
import { screen } from '@testing-library/react'

const props = {
  name: 'arabek',
  image: '',
  logo: '',
  telephone: '551143829385',
  website: 'http://www.arabek.com.br'
}

describe('<RestaurantHeader />', () => {
  it('should render correctly', () => {
    renderWithRouterAndTheme(<RestaurantHeader {...props} />)

    const backButton = screen.getByRole('img', { name: /botÃ£o de voltar/i })
    const logo = screen.getByRole('img', {
      name: /logomarca do restaurante arabek/i
    })
    const name = screen.getByRole('heading', { level: 1 })
    const telephone = screen.getByText(/551143829385/i)
    const website = screen.getByText(/www.arabek.com.br/i)

    expect(backButton).toBeInTheDocument()
    expect(logo).toBeInTheDocument()
    expect(name).toBeInTheDocument()
    expect(name).toHaveTextContent('arabek')
    expect(telephone).toBeInTheDocument()
    expect(website).toBeInTheDocument()
  })

  it('should match the snapshot', () => {
    const { container } = renderWithRouterAndTheme(
      <RestaurantHeader {...props} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
