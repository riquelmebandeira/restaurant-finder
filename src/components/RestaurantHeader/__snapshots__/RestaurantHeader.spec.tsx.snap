// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantHeader /> should match the snapshot 1`] = `
<header
  class="sc-bcXHqe jLBTwN"
>
  <div
    class="sc-gswNZR gdmCMg"
  >
    <img
      alt="botÃ£o de voltar"
      src="left-arrow.svg"
    />
    <span
      class="sc-eDvSVe deGIFn"
      color="white"
      font-size="xxsmall"
      font-weight="800"
    >
      Voltar
    </span>
  </div>
  <div
    class="sc-dkrFOg eeNFKa"
  >
    <img
      alt="logomarca do restaurante arabek"
      src=""
    />
    <div>
      <h1
        class="sc-hLBbgP ljIFmN"
      >
        arabek
      </h1>
      <p
        class="sc-eDvSVe dsvcDV"
      >
        551143829385
        <br />
        http://www.arabek.com.br
      </p>
    </div>
  </div>
</header>
`;
